docs/extend/modular-resources/create.md

```diff
@@ -0,0 +1,515 @@
+- Additionally, return any values designated in the function's return signature, typed correctly.
+
+{{% /alert %}}
+
+### Compile the module into an executable
+
+To [add a module](/extend/modular-resources/configure/#configure-your-module) to the configuration of your robot, you need to have an [executable](https://en.wikipedia.org/wiki/Executable) that runs your module when executed, can take a local socket as a command line argument, and cleanly exits when sent a termination signal.
+
+Your options for completing this step are flexible, as this file does not need to be in a raw binary format.
+
+{{< tabs >}}
+{{% tab name="Go" %}}
+
+If using the Go SDK, use Go to [compile](https://pkg.go.dev/cmd/go#hdr-Compile_packages_and_dependencies) and obtain a single executable for your module:
+
+- Navigate to your module directory in your terminal.
+- Run `go build` to compile your entry point (main program) file <file>main.go</file> and all other <file>.go</file> files in the directory, building your module and all dependencies into a single executable file.
+- Run `ls` in your module directory to find the executable, which should have been named after the module directory.
+
+Expand the [Go module code](#code-a-main-program-that-serves-as-the-entry-point-file-to-the-module) to view <file>main.go</file> for an example of this.
+<file>main.go</file> adds the custom model <file>mybase.go</file> from the resource registry, while <file>mybase.go</file> defines and registers the module.
+
+{{% /tab %}}
+{{% tab name="Python" %}}
+
+If using the Python SDK, one option is to create and save a new shell script (<file>.sh</file>) that runs your module at your entry point (main program) file.
+
+For example:
+
+``` shell
+#!/bin/sh
+cd `dirname $0`
+
+# Be sure to use `exec` so that termination signals reach the python process,
+# or handle forwarding termination signals manually
+exec python3 -m <your-module-directory-name>.<main-program-filename-without-extension> $@
+```
+
+To make this shell script executable, run the following command in your terminal:
+
+``` shell
+sudo chmod +x <FILEPATH>/<FILENAME>
+```
+
+{{% /tab %}}
+{{% /tabs %}}
+
+You need to ensure any dependencies for your module (including the Viam SDK) are installed, as well.
+Your executable will be run by `viam-server` as root, so dependencies need to be available to the root user.
+
+## Next Steps
+
+Follow [these instructions](/extend/modular-resources/configure/) to add your custom resource to your robot.
+
+Make your [executable accessible](/extend/modular-resources/configure/#make-sure-viam-server-can-access-your-executable), [configure your new module](/extend/modular-resources/configure/#configure-your-module), and [configure your modular resource](/extend/modular-resources/configure/#configure-your-modular-resource).
+
+After configuring a modular resource, test it with the [Control tab](/manage/fleet/#remote-control) and program it with the [Viam SDKs](/program/apis/).
```

Comment:
```suggestion
```
I think one next step (configure) makes sense here, and for clarity it's best to just leave it at that. Maybe add this step to the bottom of the configure page though!! I think that'd be great!

Reply:
Same as above

Reply:
I'm saying you should move this line to the bottom of the configure instructions since it's not relevant to people here yet, until after they configure. I think of the "Next steps" as being an actionable place where we guide them to exactly what to do next, and I don't want them to go try and test stuff when they can't yet.